Makefile.pre.in

```diff
@@ -801,31 +782,17 @@ Python/sysmodule.o: $(srcdir)/Python/sysmodule.c Makefile
 
 $(IO_OBJS): $(IO_H)
 
-$(PGEN): $(PGENOBJS)
-		$(CC) $(OPT) $(PY_CORE_LDFLAGS) $(PGENOBJS) $(LIBS) -o $(PGEN)
-
 .PHONY: regen-grammar
-regen-grammar: $(PGEN)
+regen-grammar:
 	# Regenerate Include/graminit.h and Python/graminit.c
-	# from Grammar/Grammar using pgen
+	# from Grammar/Grammar using pgen2
 	@$(MKDIR_P) Include
-	$(PGEN) $(srcdir)/Grammar/Grammar \
+	$(PYTHON_FOR_REGEN) -m Parser.pgen2 $(srcdir)/Grammar/Grammar \
 		$(srcdir)/Include/graminit.h.new \
 		$(srcdir)/Python/graminit.c.new
 	$(UPDATE_FILE) $(srcdir)/Include/graminit.h $(srcdir)/Include/graminit.h.new
 	$(UPDATE_FILE) $(srcdir)/Python/graminit.c $(srcdir)/Python/graminit.c.new
 
-Parser/grammar.o:	$(srcdir)/Parser/grammar.c \
-				$(srcdir)/Include/token.h \
-				$(srcdir)/Include/grammar.h
-Parser/metagrammar.o:	$(srcdir)/Parser/metagrammar.c
-
-Parser/tokenizer_pgen.o:	$(srcdir)/Parser/tokenizer.c
-Parser/parsetok_pgen.o:	$(srcdir)/Parser/parsetok.c
```

Comment:
If you delete parsetok.c. how do you generate parsetok.o on line:
https://github.com/python/cpython/blob/6d06fd8788e27615c33d0ef0df3a845170c35f84/Makefile.pre.in#L301
?

Reply:
`parsetok_pgen` is the one that is not needed. The source `parsetok.c` is still on the source tree so the makefile generates `parsetok.o` because is still a target.